### PR TITLE
Checkout: Update color themes to use WordPress blue colors

### PR DESCRIPTION
**Pull request by sirbrillig**
## Background

I was [researching](https://github.com/Automattic/payments-shilling/issues/3060) why the buttons in checkout are a different shade of blue than other parts of calypso.

For example, see the color difference between these two buttons:

Outside checkout             |  Checkout
:-------------------------:|:-------------------------:
Screenshot 2024-09-03 at 10 45 43 AM | Screenshot 2024-09-03 at 10 59 29 AM

It turns out that checkout is [using the color directly](https://github.com/Automattic/wp-calypso/blob/66c1aea529d734ee3cab9f5a1a877bd2ee5b963e/packages/composite-checkout/src/lib/theme.ts#L63) from the `@automattic/color-studio` package (which [is outdated as version 2.6](https://github.com/Automattic/wp-calypso/blob/66c1aea529d734ee3cab9f5a1a877bd2ee5b963e/package.json#L151) everywhere in calypso) but some parts of calypso are using the [deprecated](https://github.com/Automattic/wp-calypso/blob/66c1aea529d734ee3cab9f5a1a877bd2ee5b963e/packages/components/src/button/index.tsx#L107-L111) button in the @automattic/components package which has [a hard-coded color](https://github.com/Automattic/wp-calypso/blob/66c1aea529d734ee3cab9f5a1a877bd2ee5b963e/packages/calypso-color-schemes/src/shared/color-schemes/_global.scss#L14) (from the `@automattic/calypso-color-schemes` package) that happens to match the one in color-studio 3.0. In https://github.com/Automattic/wp-calypso/pull/94148 I'm updating all of calypso to use the latest version of the color-studio package.

However, checkout currently [uses the color](https://github.com/Automattic/color-studio/blob/trunk/dist/colors.json#L26) `Blue 50` for its buttons, and [the blue used by the "correct" color scheme](https://github.com/Automattic/color-studio/blob/trunk/dist/colors.json#L143) is actually `WordPress Blue 50`. So to get the correct color in checkout we need to also update the `@automattic/composite-checkout` package to be use `WordPress Blue` instead.

## Proposed changes

In this PR I'm updating the colors used by the `@automattic/composite-checkout` package theme (and the Jetpack variant) to use "WordPress Blue" instead of "Blue".

## Screenshot

This change by itself:

Before             |  After
:-------------------------:|:-------------------------:
Screenshot 2024-09-03 at 11 29 05 AM | Screenshot 2024-09-03 at 11 59 33 AM

With https://github.com/Automattic/wp-calypso/pull/94148 also applied:

Before             |  After
:-------------------------:|:-------------------------:
Screenshot 2024-09-03 at 11 29 05 AM | Screenshot 2024-09-03 at 11 43 27 AM

## Testing Instructions

- Add a product to your cart and visit checkout.
- If the steps are collapsed (auto-completed), click to "Edit" the tax information step ("Billing details" or "Contact details").
- Examine the color of the "Continue to payment" button and verify it is `#006088` (previously`#0675c4`).
- Optional: also apply https://github.com/Automattic/wp-calypso/pull/94148 and verify that the color has changed to `#3858e9`.

### CODE REVIEW

**Review by michaeldcain — APPROVED** [review]
This is good to go once #94148 is merged. Let's not merge it before then, so as to not introduce another blue to our primary flows (Checkout currently matches Signup and Stepper, but not Calypso or Sites Dashboard).

**Review by fditrapani — APPROVED** [review]
Thanks for this update. This PR is good to go. 

We'll need to make a subsequent update that updates the blue for the following:
- All text links should have the new blue (In terms, the support link, and edit button)
- The title for the callout in the sidebar
- The active state for the dropdown options in the term
- The borders for the payment methods (both hover and active state)

Also, unrelated, I noticed the borders between the payment method have a rounded edge on them when unselected:
image

**Pull request merged**
